packages/template-retail-react-app/app/hooks/use-password-reset.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {AuthHelpers, useAuthHelper} from '@salesforce/commerce-sdk-react'
+import {useToast} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {useIntl} from 'react-intl'
+import {absoluteUrl} from '@salesforce/retail-react-app/app/utils/url'
+/**
+ * This hook provides commerce-react-sdk hooks to simplify the reset password flow.
+ */
+export const usePasswordReset = () => {
+    const toast = useToast()
+    const {formatMessage} = useIntl()
+
+    const getPasswordResetTokenMutation = useAuthHelper(AuthHelpers.GetPasswordResetToken)
+    const resetPasswordMutation = useAuthHelper(AuthHelpers.ResetPassword)
+
+    const getPasswordResetToken = async (email) => {
+        await getPasswordResetTokenMutation.mutateAsync({
+            user_id: email,
+            callback_uri: absoluteUrl('/reset-password-callback')
```

Comment:
NOTE: If customers wanted to customize the callback URI they would have to come into this hook and change this line... Or should we have this configurable in the config file ?🤔

Reply:
Warning: absoluteUrl is likely to be deprecated due to the use of getOrigin. Please use [useAppOrigin](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/template-retail-react-app/app/hooks/use-app-origin.js) for the url building.